apps/heft/src/cli/HeftActionRunner.ts

```diff
@@ -37,14 +38,36 @@ import type {
   IHeftLifecycleToolStartHookOptions
 } from '../pluginFramework/HeftLifecycleSession';
 import type { HeftLifecycle } from '../pluginFramework/HeftLifecycle';
-import type { HeftTask } from '../pluginFramework/HeftTask';
+import type { IHeftTask, HeftTask } from '../pluginFramework/HeftTask';
 import { deleteFilesAsync, type IDeleteOperation } from '../plugins/DeleteFilesPlugin';
 import { Constants } from '../utilities/Constants';
 
 export interface IHeftActionRunnerOptions extends IHeftActionOptions {
   action: IHeftAction;
 }
 
+/**
+ * Metadata for an operation that represents a task.
+ * @public
+ */
+export interface IHeftTaskOperationMetadata {
+  task: IHeftTask;
+  phase: IHeftPhase;
+}
+
+/**
+ * Metadata for an operation that represents a phase.
+ * @public
+ */
+export interface IHeftPhaseOperationMetadata {
+  phase: IHeftPhase;
+}
+
+/**
+ * @internal
+ */
+export type HeftOperationMetadata = IHeftTaskOperationMetadata | IHeftPhaseOperationMetadata;
```

Comment:
I think this would get a lot simpler if we treat the "phase" operation as a synthetic task (from a metadata perspective), then we can use the same metadata type for everything.